src/data/sonata_data.cpp

```diff
@@ -262,36 +258,50 @@ void SonataData::write_report_header() {
     hdf5_writer_->configure_attribute(reports_population_group + "/data", "units", report_units_);
 
     std::vector<uint64_t> sonata_node_ids(node_ids_);
-    convert_gids_to_sonata(sonata_node_ids);
+    convert_gids_to_sonata(sonata_node_ids, population_offset_);
     hdf5_writer_->write(reports_population_group + "/mapping/node_ids", sonata_node_ids);
     hdf5_writer_->write(reports_population_group + "/mapping/index_pointers", index_pointers_);
     hdf5_writer_->write(reports_population_group + "/mapping/element_ids", element_ids_);
     hdf5_writer_->write_time(reports_population_group + "/mapping/time", time_);
     hdf5_writer_->configure_attribute(reports_population_group + "/mapping/time", "units", "ms");
 }
 
-void SonataData::write_spikes_header(const std::string& order_by) {
+void SonataData::write_spikes_header(const std::unique_ptr<Population>& population) {
```

Comment:
It's typically a bad practice to pass around references to pointers. Since we are not changing the pointer nor the object, I would just pass the reference to the object directly. In the caller function it will just require dereferencing the pointer with `*`
```suggestion
void SonataData::write_spikes_header(const Population& population) {
```

Reply:
Done